service/history/execution/mutable_state_builder.go

```diff
@@ -1800,7 +1804,15 @@ func (e *mutableStateBuilder) ReplicateWorkflowExecutionStartedEvent(
 ) error {
 
 	event := startEvent.WorkflowExecutionStartedEventAttributes
+	if event.GetRequestID() != "" {
+		requestID = event.GetRequestID()
```

Comment:
add a comment about why we are overriding the `requestID` passed with the event's request id. what each of those would mean